src/core/async-work-cache.ts

```diff
@@ -31,11 +35,34 @@ export class AsyncWorkCache<K, V> {
    *
    * If not, compute it with the given function.
    *
-   * This method ensures that we will only try to compute the value for `key`
-   * once, no matter how often or with what timing getOrCompute is called, even
-   * recursively.
+   * This method ensures that, in the absence of cancellations, we will only try
+   * to compute the value for `key` once, no matter how often or with what
+   * timing getOrCompute is called, even recursively.
+   *
+   * If a compute() throws a Cancel that should not result in a Cancel for
+   * non-cancelled operations. So long as the given cancelToken is not
+   * cancelled, the caller will not receive a Cancel.
    */
-  async getOrCompute(key: K, compute: () => Promise<V>) {
+  async getOrCompute(
+      key: K, compute: () => Promise<V>, cancelToken?: CancelToken) {
+    cancelToken = cancelToken || neverCancels;
```

Comment:
put the default in the parameter definition

Reply:
Ah, yeah, that's something we can do now that we've dropped node v4.